lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -318,12 +319,61 @@ def _on_list_backups_action(self, event) -> None:
             return
 
         try:
-            backups = subprocess.check_output("percona-backup-mongodb list", shell=True)
-            event.set_results({"backups": backups.decode("utf-8")})
+            formatted_list = self._generate_backup_list_output()
+            event.set_results({"backups": formatted_list})
         except subprocess.CalledProcessError as e:
             event.fail(f"Failed to list MongoDB backups with error: {str(e)}")
             return
 
+    def _generate_backup_list_output(self) -> str:
+        """Generates a list of backups in a formatted table.
+
+        List contains successful, failed, and in progress backups in order of ascending time.
+
+        Raises CalledProcessError
+        """
+        backup_list = []
+        pbm_status = subprocess.check_output(
+            "percona-backup-mongodb status --out=json", shell=True, stderr=subprocess.STDOUT
+        )
+        # processes finished and failed backups
+        pbm_status = json.loads(pbm_status.decode("utf-8"))
+        backups = pbm_status["backups"]["snapshot"] or []
+        for backup in backups:
+            backup_status = "finished"
+            if backup["status"] == "error":
+                backup_status = "failed"
+            if backup["status"] != "error" and backup["status"] != "done":
+                backup_status = "in progress"
+            backup_list.append((backup["name"], backup["type"], backup_status))
+
+        # process in progress backups
+        running_backup = pbm_status["running"]
+        if running_backup.get("type", None) == "backup":
+            # backups are sorted in reverse order
+            last_reported_backup = backup_list[0]
+            # pbm will occasionally report backups that are currently running as failed, so it is
+            # necessary to correct the backup list in this case.
+            if last_reported_backup[0] == running_backup["name"]:
+                backup_list[0] = (last_reported_backup[0], last_reported_backup[1], "in progress")
+            else:
+                backup_list.append((running_backup["name"], "logical", "in progress"))
```

Comment:
Is the backup type always logical in this case?

Reply:
Yes, until we support physical backups any backup running on PBM will be logical :)